File: share/src/main/java/com/uneed/yuni/wxapi/WXEntryActivity.java
package com.uneed.yuni.wxapi;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;
import android.os.Message;
import android.util.Log;

import com.tencent.mm.opensdk.constants.ConstantsAPI;
import com.tencent.mm.opensdk.modelbase.BaseReq;
import com.tencent.mm.opensdk.modelbase.BaseResp;
import com.tencent.mm.opensdk.openapi.IWXAPI;
import com.tencent.mm.opensdk.openapi.IWXAPIEventHandler;
import com.tencent.mm.opensdk.openapi.WXAPIFactory;
import com.tikoua.share.wechat.WXConst;
import com.tikoua.share.wechat.WechatShareMeta;
import com.tikoua.share.wechat.WechatUtilKt;

import java.lang.ref.WeakReference;

public class WXEntryActivity extends Activity implements IWXAPIEventHandler {
    private static String TAG = "MicroMsg.WXEntryActivity";

    private IWXAPI api;
    private MyHandler handler;

    private static class MyHandler extends Handler {
        private final WeakReference<WXEntryActivity> wxEntryActivityWeakReference;

        public MyHandler(WXEntryActivity wxEntryActivity) {
            wxEntryActivityWeakReference = new WeakReference<WXEntryActivity>(wxEntryActivity);
        }

        @Override
        public void handleMessage(Message msg) {
            int tag = msg.what;
            Log.d("MyHandler", "handleMessage tag: " + tag);
            /*switch (tag) {
                case NetworkUtil.GET_TOKEN: {
                    Bundle data = msg.getData();
                    JSONObject json = null;
                    try {
                        json = new JSONObject(data.getString("result"));
                        String openId, accessToken, refreshToken, scope;
                        openId = json.getString("openid");
                        accessToken = json.getString("access_token");
                        refreshToken = json.getString("refresh_token");
                        scope = json.getString("scope");
                        Intent intent = new Intent(wxEntryActivityWeakReference.get(), SendToWXActivity.class);
                        intent.putExtra("openId", openId);
                        intent.putExtra("accessToken", accessToken);
                        intent.putExtra("refreshToken", refreshToken);
                        intent.putExtra("scope", scope);
                        wxEntryActivityWeakReference.get().startActivity(intent);
                    } catch (JSONException e) {
                        Log.e(TAG, e.getMessage());
                    }
                }
            }*/
        }
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        Log.d("MyHandler", "onCreate");
        WechatShareMeta wechatMeta = WechatUtilKt.getWechatMeta(this);
        api = WXAPIFactory.createWXAPI(this, wechatMeta.getAppid(), false);
        handler = new MyHandler(this);

        try {
            Intent intent = getIntent();
            api.handleIntent(intent, this);
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    @Override
    protected void onNewIntent(Intent intent) {
        super.onNewIntent(intent);
        Log.d("MyHandler", "onNewIntent");
        setIntent(intent);
        api.handleIntent(intent, this);
    }

    /**
     * 微信发送的请求将回调到 onReq 方法
     *
     * @param req
     */
    @Override
    public void onReq(BaseReq req) {
        Log.d("MyHandler", "onReq");
        switch (req.getType()) {
            case ConstantsAPI.COMMAND_GETMESSAGE_FROM_WX:
                break;
            case ConstantsAPI.COMMAND_SHOWMESSAGE_FROM_WX:
                break;
            default:
                break;
        }
        finish();
    }

    /**
     * 发送到微信请求的响应结果将回调到 onResp 方法
     *
     * @param resp
     */
    @Override
    public void onResp(BaseResp resp) {
        int errCode = resp.errCode;
        int type = resp.getType();
        Log.d("MyHandler", "onResp  errCode: " + errCode + "  type: " + type);
        Intent intent = new Intent(WXConst.ActionWXResp);
        intent.putExtra("ec", errCode);
        sendBroadcast(intent);
        finish();
    }
}

File: app/src/main/java/com/uneed/yuni/wxapi/WXEntryActivity.kt
package com.uneed.yuni.wxapi

import com.tikoua.share.utils.log
import com.tikoua.share.wechat.WechatHandlerActivity

class WXEntryActivity : WechatHandlerActivity() {
    /**
     * 处理微信发出的向第三方应用请求app message
     *
     *
     * 在微信客户端中的聊天页面有“添加工具”，可以将本应用的图标添加到其中
     * 此后点击图标，下面的代码会被执行。Demo仅仅只是打开自己而已，但你可
     * 做点其他的事情，包括根本不打开任何页面
     */
    override fun onGetMessageFromWXReq(msg: Any?) {
        log("onGetMessageFromWXReq: $msg")
        if (msg != null) {
            val iLaunchMyself =
                packageManager.getLaunchIntentForPackage(packageName)
            startActivity(iLaunchMyself)
        }
    }

    /**
     * 处理微信向第三方应用发起的消息
     *
     *
     * 此处用来接收从微信发送过来的消息，比方说本demo在wechatpage里面分享
     * 应用时可以不分享应用文件，而分享一段应用的自定义信息。接受方的微信
     * 客户端会通过这个方法，将这个信息发送回接收方手机上的本demo中，当作
     * 回调。
     *
     *
     * 本Demo只是将信息展示出来，但你可做点其他的事情，而不仅仅只是Toast
     */
    override fun onShowMessageFromWXReq(msg: Any?) {
        log("onShowMessageFromWXReq: $msg")
        if (msg != null) {
            val iLaunchMyself =
                packageManager.getLaunchIntentForPackage(packageName)
            startActivity(iLaunchMyself)
        }
    }
}